src/pretix/base/models/discount.py

```diff
@@ -355,11 +368,15 @@ def apply(self, positions: Dict[int, PositionInfo],
         # First, filter out everything not even covered by our product scope
         condition_candidates = [
             idx
-            for idx, (item_id, subevent_id, line_price_gross, is_addon_to, voucher_discount) in positions.items()
+            for idx, (item_id, subevent_id, subevent_date_from, line_price_gross, is_addon_to, voucher_discount) in
+            positions.items()
             if (
                 (self.condition_all_products or item_id in limit_products) and
                 (self.condition_apply_to_addons or not is_addon_to) and
                 (not self.condition_ignore_voucher_discounted or voucher_discount is None or voucher_discount == Decimal('0.00'))
+                and (not subevent_id or (
+                    self.event_date_from is None or self.event_date_from and subevent_date_from >= self.event_date_from)) and (
+                        self.event_date_until is None or self.event_date_until and subevent_date_from < self.event_date_until)
```

Comment:
I think this can be reduced to 

```suggestion
                    self.event_date_from is None or subevent_date_from >= self.event_date_from)) and (
                        self.event_date_until is None or subevent_date_from < self.event_date_until)
```

or what am I missing?

Reply:
Not missing anything, as far as I can tell, just a leftover snip from sometime earlier that is now removed